src/zarr/api/asynchronous.py

```diff
@@ -691,22 +730,31 @@ async def open_group(
     if chunk_store is not None:
         warnings.warn("chunk_store is not yet implemented", RuntimeWarning, stacklevel=2)
 
-    store_path = await make_store_path(store, mode=mode, storage_options=storage_options, path=path)
+    store_mode = _handle_store_mode(mode)
+    exists_ok = _infer_exists_ok(mode)
+    store_path = await make_store_path(
+        store, mode=store_mode, storage_options=storage_options, path=path
+    )
+    await store_path._init(mode=mode)
 
     if attributes is None:
         attributes = {}
 
     try:
-        return await AsyncGroup.open(
-            store_path, zarr_format=zarr_format, use_consolidated=use_consolidated
-        )
+        if mode in {"r", "r+", "a"}:
+            return await AsyncGroup.open(
+                store_path, zarr_format=zarr_format, use_consolidated=use_consolidated
+            )
     except (KeyError, FileNotFoundError):
+        pass
+    if mode in {"a", "w", "w-"}:
```

Comment:
should we bind this set to a variable, and similarly for `{"r", "r+", "a"}`? that would improve the literacy a bit.

Reply:
`READ_MODES` and `WRITE_MODES`?

Reply:
works for me